flepimop/gempyor_pkg/src/gempyor/_jinja.py

```diff
@@ -31,15 +31,21 @@ def _render_template_to_file(name: str, data: dict[str, Any], file: Path) -> Non
         data: The data to pass to the template when rendering.
         file: The file to save the rendered template to.
 
+    Returns:
+        The `file` argument given.
+
     Examples:
+        >>> from gempyor._jinja import _render_template_to_file
         >>> from pathlib import Path
         >>> file = Path("hi.txt")
         >>> _render_template_to_file("test_template.j2", {"name": "Jane"}, file)
+        PosixPath('hi.txt')
         >>> file.read_text()
         'Hello Jane!'
     """
     with file.open(mode="w", encoding="utf-8") as f:
         f.write(_jinja_environment.get_template(name).render(data))
+    return file
```

Comment:
hmm - should the return be within the `with` block or not? cursory googling suggests mixed opinions.

Reply:
also, either way: the code in the next method (`_render_template_to_temp_file`) can now also return this invocation directly, yes?

Reply:
I've amended the commit to include the change to `_render_template_to_temp_file`. I've generally not put a return within the context manager, I think that can have edge cases around getting properties of the file that don't apply here.